src/App.js

```diff
@@ -9,16 +14,28 @@ import './App.css';
 import List from './pages/List/List';
 import AddItem from './pages/AddItem/AddItem';
 import NotFound from './pages/NotFound/NotFound';
+import Home from './pages/Home/Home';
 
 // Components
 import NavMenu from './components/NavMenu/NavMenu';
 
 function App() {
+  const [haveToken, setHaveToken] = useState(false);
+
+  useEffect(() => {
+    const token = localStorage.getItem('token');
+    if (token) {
+      setHaveToken(true);
+    }
+  }, []);
   return (
     <Router>
       <div className="App container">
         <Switch>
           <Route exact path="/">
+            {haveToken ? <Redirect to="/list" /> : <Home />}
+          </Route>
+          <Route path="/list">
```

Comment:
@anderswift could you elaborate on how we would add a check here? Would we route the user conditional based on if the list ID exists? If it does exist, would you still want to pass the list ID down to our form?

Reply:
@connietran-dev As is, the ShoppingList and AddItemForm components you and I created ([in this PR](https://github.com/the-collab-lab/tcl-26-smart-shopping-list/pull/21)) to read and write to the shopping list rely on a listId (listId is our a state variable where we store the document ID of a specific "list" in Firestore). One way to ensure those components work would be to prevent the user from visiting the "Add Item" or "List" views that use those components unless a list ID has been set. So yes, routing conditional on whether the list ID exists. 

We should probably also consider that a listId would not have been set in the following situations, and make sure we are accounting for them (and providing feedback to the user where necessary):

- when App first loads and hasn't had a chance to look up a list ID yet
- if the user doesn't have a token set yet (new user, or has cleared cache), so can't retrieve an associated list ID—in this case we'd want to show the Home view and let them create a new token (or eventually provide a shared token)
- if the user \*does\* have a token in localStorage, but there is a problem connecting to the database
- if the user \*does\* have a token in localStorage, but it's not valid (no associated list in the database can be found)

Reply:
@anderswift Yes, we can use `<Redirect />`, as we have in our PR, to control where the user is sent for a given route based on a condition. I believe what you are referencing here is a scenario where the user manually goes to the /list or /add URLs without having added any list? In that case, we can use `<Redirect />` on each to check the relevant condition and then I assume route them back to the home page. Per my comment above, I think for the most part that condition will be `listId`. 

Related to your further considerations (in the bulleted list), I think we should handle those after we merge for this week to avoid trying to do too much at one time before we are reorganized in a common branch again. Those are all definitely things we need to address, thank you for thinking of them!

Reply:
Sounds like a good plan to me!